client/blocks/signup-form/index.jsx

```diff
@@ -194,9 +196,16 @@ class SignupForm extends Component {
 	/**
 	 * If the step is invalid because we had an error that the user exists,
 	 * we should prompt user with a request to connect his social account
-	 * to his existing WPCOM account
+	 * to his existing WPCOM account.
+	 *
+	 * That can be done either by redirecting or only suggesting. If it's done
+	 * by suggesting, bail out of redirecting and display the error.
 	 */
 	maybeRedirectToSocialConnect() {
```

Comment:
With the new behavior, will we ever redirect to social connect? I might misunderstand, but it seem like the two options now are "create account with different method" or "log in with the same email," but not social connection. 

If that's the case, maybe we can remove this code?

Reply:
> With the new behavior, will we ever redirect to social connect? I might misunderstand, but it seem like the two options now are "create account with different method" or "log in with the same email," but not social connection.
> 
> If that's the case, maybe we can remove this code?

@noahtallen I'm not sure, to be honest ☹️ 

@p-jackson Any idea? Or do we need to wait until @zaguiini is back?

Reply:
Cool! Is it easy for this new behavior to apply to those other uses of SignupForm, or are those isolated enough that it wouldn't work?

Reply:
> Is it easy for this new behavior to apply to those other uses of SignupForm, or are those isolated enough that it wouldn't work?

@noahtallen I _think_ so, but I'm not confident it's expected. We've already run outside the initial scope of the project, so I'm not sure we want to spend more time on it.

Reply:
Fair enough! :)